src/solver/infeasible-problem-analysis/watched-constraints.cpp

```diff
@@ -0,0 +1,217 @@
+}
+
+std::string WeeklyBC::infeasibilityCause()
+{
+    return "* Weekly binding constraints.";
+}
+
+// --- Fictitious load constraint ---
+std::string FictitiousLoad::infeasibility()
+{
+    return "Last resort shedding status at area '" + areaName(splitName()) + "' at hour "
+           + timeStep(splitName());
+}
+
+std::string FictitiousLoad::infeasibilityCause()
+{
+    return "* Last resort shedding status.";
+}
+
+// --- Hydro level constraint ---
+std::string HydroLevel::infeasibility()
+{
+    return "Hydro level constraint at area '" + areaName(splitName()) + "' at hour "
+           + timeStep(splitName());
+}
+
+std::string HydroLevel::infeasibilityCause()
+{
+    return "* Hydro reservoir impossible to manage with cumulative options "
+           "\"hard bounds without heuristic\"";
+}
+
+// --- Short term storage constraint ---
+std::string STS::infeasibility()
+{
+    return "Short-term-storage reservoir constraint at area '" + areaName(splitName())
+           + "' in STS '" + STSname(splitName()) + "' at hour " + timeStep(splitName());
+}
+
+std::string STS::infeasibilityCause()
+{
+    return "* Short-term storage reservoir level impossible to manage. Please check inflows, "
+           "lower & upper curves and initial level (if prescribed),";
+}
+
+// --- Hydro production constraint ---
+std::string HydroProduction::infeasibility()
+{
+    return "Hydro weekly production at area '" + areaName(splitName()) + "'";
+}
+
+std::string HydroProduction::infeasibilityCause()
+{
+    return "* impossible to generate exactly the weekly hydro target";
+}
+
+// --- Constraints factory ---
+std::shared_ptr<WatchedConstraint> createHourlyBC(std::string varName)
+{
+    return std::make_shared<HourlyBC>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createDailyBC(std::string varName)
+{
+    return std::make_shared<DailyBC>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createWeeklyBC(std::string varName)
+{
+    return std::make_shared<WeeklyBC>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createFictitiousLoad(std::string varName)
+{
+    return std::make_shared<FictitiousLoad>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createHydroLevel(std::string varName)
+{
+    return std::make_shared<HydroLevel>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createSTS(std::string varName)
+{
+    return std::make_shared<STS>(varName);
+}
+
+std::shared_ptr<WatchedConstraint> createHydroProduction(std::string varName)
+{
+    return std::make_shared<HydroProduction>(varName);
+}
+
+ConstraintsFactory::ConstraintsFactory()
+{
+    regex_to_ctypes_["::hourly::"] = createHourlyBC;
+    regex_to_ctypes_["::daily::"] = createDailyBC;
+    regex_to_ctypes_["::weekly::"] = createWeeklyBC;
+    regex_to_ctypes_["^FictiveLoads::"] = createFictitiousLoad;
+    regex_to_ctypes_["^AreaHydroLevel::"] = createHydroLevel;
+    regex_to_ctypes_["^Level::"] = createSTS;
+    regex_to_ctypes_["^HydroPower::"] = createHydroProduction;
```

Comment:
```suggestion
    regex_to_ctypes_["::hourly::"] = std::make_shared<HourlyBC>;
    regex_to_ctypes_["::daily::"] = //;
    regex_to_ctypes_["::weekly::"] = //;
    regex_to_ctypes_["^FictiveLoads::"] = //;
    regex_to_ctypes_["^AreaHydroLevel::"] = //;
    regex_to_ctypes_["^Level::"] = //;
    regex_to_ctypes_["^HydroPower::"] = //;
```

Reply:
This isn't what we want : 
We don't want to associate a regex to an instance of a watched constraint (for example of **HourlyBC**), but to the type of the watched constraint : for example, associate the type **HourlyBC** to "::hourly::"